pkgs/applications/networking/mailreaders/evolution/evolution-ews/hardcode-gsettings.cocci

```diff
@@ -0,0 +1,17 @@
+@@
```

Comment:
It would be nice to re-use the patch from e-d-s, since it has more checking. Possibly with something like the following to only whitelist required schemas per each package.

```patch
diff --git a/pkgs/desktops/gnome/core/evolution-data-server/default.nix b/pkgs/desktops/gnome/core/evolution-data-server/default.nix
index a7d4673158b..d7fd7e08e70 100644
--- a/pkgs/desktops/gnome/core/evolution-data-server/default.nix
+++ b/pkgs/desktops/gnome/core/evolution-data-server/default.nix
@@ -147,7 +147,9 @@ stdenv.mkDerivation rec {
           cd "''${sourceRoot:-.}"
           git init
           git add -A
-          spatch --sp-file "${./hardcode-gsettings.cocci}" --dir . --in-place
+          cp ${./hardcode-gsettings.cocci} hardcode-gsettings.cocci
+          cp ${./run.cocci} run.cocci
+          spatch --sp-file "run.cocci" --dir . --in-place
           git diff > "$out"
         '';
 
diff --git a/pkgs/desktops/gnome/core/evolution-data-server/hardcode-gsettings.cocci b/pkgs/desktops/gnome/core/evolution-data-server/hardcode-gsettings.cocci
index 9e60e892b71..1d31f686886 100644
--- a/pkgs/desktops/gnome/core/evolution-data-server/hardcode-gsettings.cocci
+++ b/pkgs/desktops/gnome/core/evolution-data-server/hardcode-gsettings.cocci
@@ -16,15 +16,11 @@ def register_cpp_constant(const_name, val):
 def resolve_cpp_constant(const_name):
     return cpp_constants.get(const_name, const_name)
 
+# These are supposed to be replaced by run.cocci
 e_s_d_schema_constants = [
-    # The following are actually part of e-d-s, despite the name.
-    # We rename the old ambiguos constant name in ./prepare-for-gsettings-patching.patch
-    "\"org.gnome.Evolution.DefaultSources\"",
-    "\"org.gnome.evolution.shell.network-config\"",
 ]
 
 g_d_s_schema_constants = [
-    "\"org.gnome.system.proxy\"",
 ]
 
 def get_schema_directory(schema_path):
diff --git a/pkgs/desktops/gnome/core/evolution-data-server/run.cocci b/pkgs/desktops/gnome/core/evolution-data-server/run.cocci
new file mode 100644
index 00000000000..e8cb3abbfab
--- /dev/null
+++ b/pkgs/desktops/gnome/core/evolution-data-server/run.cocci
@@ -0,0 +1,14 @@
+#include "hardcode-gsettings.cocci"
+
+@initialize:python@
+@@
+e_s_d_schema_constants = [
+    # The following are actually part of e-d-s, despite the name.
+    # We rename the old ambiguos constant name in ./prepare-for-gsettings-patching.patch
+    "\"org.gnome.Evolution.DefaultSources\"",
+    "\"org.gnome.evolution.shell.network-config\"",
+]
+
+g_d_s_schema_constants = [
+    "\"org.gnome.system.proxy\"",
+]
```

Ideally we would move the `"\"org.gnome.evolution-data-server` prefix there but not sure about the cleanest way. Maybe having the `run.cocci` set a `schema_constants: List[Tuple[str, function]]` variable and the `hardcode-gsettings.patch` would just iterate over it.

Reply:
what is the benefit of whitelisting schemas?  It already took months to notice we were missing schema-related patches, if new schemas are introduced upstream we will almost surely miss them.

Reply:
That’s precisely the point, the patch will fail to generate if we omit a necessary schema from the whitelist.

Reply:
mmh looks like I don't understand the python part of the coccinelle patch. feel free to force push on my branch, I ticked the box.

Reply:
done